src/store/useTaskStore.ts

```diff
@@ -40,4 +43,14 @@ export const useTaskStore = create<TaskStore>((set) => ({
     }),
 
   clearTasks: () => set({ tasks: [] }),
+
+  setFilter: (filter) => set({ filter }),
+
+  // Метод для отриманя відфілтрованних завдань
+  filteredTasks: () => {
```

Comment:
Idea is right but please use selectors for it https://zustand.docs.pmnd.rs/guides/auto-generating-selectors maybe it fits better for getting filtered items